python-package/lightgbm/callback.py

```diff
@@ -363,11 +377,14 @@ def _init(self, env: CallbackEnv) -> None:
                     raise ValueError("Must provide a single value for min_delta or as many as metrics.")
                 if self.first_metric_only and self.verbose:
                     _log_info(f"Using only {self.min_delta[0]} as early stopping min_delta.")
+                if (self.chosen_metric is not None) and self.verbose:
+                    index_chosen_metric = list_metrics.index(self.chosen_metric)
```

Comment:
I don't think this works, `list_metrics` is not actually a list :eyes:

Reply:
Will look into it.
Following @jameslamb comment, this part of the code will be impacted so I will need to rewrite it anyway